### DISCUSSION TITLE
ValueError: key must be PdfObject

### DISCUSSION THREAD

**Discussion by aptara-pawan** [py-pdf/pypdf#2028] [General]
i am getting error while adding checkbox in PDF.

ValueError: key must be PdfObject

Below is my code, Please help me on this

# import chardet
# import PyPDF2
# import re

from PyPDF2 import PdfFileWriter, PdfFileReader
from PyPDF2.generic import TextStringObject, BooleanObject, NameObject, DictionaryObject, ArrayObject,PdfObject 

def add_checkbox_to_page(page, x, y, width, height):
    checkbox = {
        "/Type": "/Annot",
        "/Subtype": "/Widget",
        "/FT": "/Btn",
        "/T": TextStringObject("Checkbox"),
        "/Rect": [x, y, x + width, y + height],
        "/V": TextStringObject("Yes"),
        "/AP": DictionaryObject(
            {
                "/N": DictionaryObject({NameObject("/Off"): TextStringObject("[]"), NameObject("/Yes"): TextStringObject("[/On]")}),
                "/D": DictionaryObject({NameObject("/Off"): TextStringObject("0 0 R"), NameObject("/Yes"): TextStringObject("1 0 R")}),
            }
        ),
        "/AS": NameObject("/Yes"),
        "/H": TextStringObject("N"),
    }
    #page[NameObject("/Annots")].append(DictionaryObject(checkbox))
    page[NameObject("/Annots")] = ArrayObject([DictionaryObject(checkbox)])

def add_checkboxes_to_pdf(input_pdf, output_pdf):
    pdf_reader = PdfFileReader(input_pdf)
    pdf_writer = PdfFileWriter()

    for page_number in range(len(pdf_reader.pages)):
        page = pdf_reader.pages[page_number]
        add_checkbox_to_page(page, x=100, y=600, width=20, height=20)  # Adjust the coordinates as needed for checkbox position
        pdf_writer.addPage(page)

    with open(output_pdf, "wb") as output_file:
        pdf_writer.write(output_file)

    # pdf_output_file = open("D:\\Pawan\\temp\\25\\check\\AJC26740OUT.pdf", 'wb')
    # pdf_writer.write(pdf_output_file)

# Call the function to add checkboxes to the PDF
add_checkboxes_to_pdf("D:\\Pawan\\temp\\25\\check\\test.pdf", "D:\\Pawan\\temp\\25\\check\\AJC26740_OUT.pdf")

**Comment by pubpub-zz** [comment]
This is already addressed in an issue.
